### PR TITLE
Add some documentation on shared forms

### PR DESCRIPTION
**Pull request by dwhenry**
## Description of change
Add some documentation for shared forms in the engine

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRM457-176

### CODE REVIEW

File: gems/laa_multi_step_forms/docs/SharedForms.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+This form is designed to be used when it is possible to add multiple instance of
+an object, and the index page asks the question `Do you want to add another`.
+
+The form has a single YesNoAnswer attribute and does not modify the application
+when processed.
+
+When using this pattern:
+*  the controller will need to load any other data need for
+the view into a separate data variable, this should be done in both thge `edit`
```
[the remark this answers is not in the document]
Comment (ivanELEC) [reply gems/laa_multi_step_forms/docs/SharedForms.md:16]: also is this meant to read "any other data needed"?

File: gems/laa_multi_step_forms/docs/SharedForms.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    def update
+      @data_for_view = current_application.data_for_view
+      update_and_advance(AddAnotherForm, as: :work_items)
+    end
+
+    private
+
+    def decision_tree_class
+      Decisions::SimpleDecisionTree
+    end
```
Comment (ivanELEC) [inline gems/laa_multi_step_forms/docs/SharedForms.md:42]: Like adding in the additional_permitted_params, what is this decision_tree_class added in as a method for? Is this another example of black box behaviour?

File: gems/laa_multi_step_forms/docs/SharedForms.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+attribute (to track which sub object should be deleted) and destorys the record
+when processed.
+
+When using this pattern:
+* when determining the specific item to be deleted the id is passed in different
+ways from the `edit` and `update` endpoints. (TODO: review how this can be
+improved, potentially with nested routes)
+* a flash message should be passed on success
+* a fallback if the record can't be found should redirect the user to the index
+page (for the set fo objects)
```
[the remark this answers is not in the document]
Comment (dwhenry) [reply gems/laa_multi_step_forms/docs/SharedForms.md:64]: fixed

File: gems/laa_multi_step_forms/docs/SharedForms.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dwhenry) [reply gems/laa_multi_step_forms/docs/SharedForms.md:16]: fixed

File: gems/laa_multi_step_forms/docs/SharedForms.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
+# Shared forms
+
+These are form objects that are generic enough the have been added into the gem
+and can be reused in specific situations
+
+## AddAnotherForm
```
[the remark this answers is not in the document]
Comment (dwhenry) [reply gems/laa_multi_step_forms/docs/SharedForms.md:6]: updated

File: gems/laa_multi_step_forms/docs/SharedForms.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+    private
+
+    def decision_tree_class
+      Decisions::SimpleDecisionTree
+    end
+
+    def additional_permitted_params
+      [:add_another]
+    end
```
[the remark this answers is not in the document]
Comment (dwhenry) [reply gems/laa_multi_step_forms/docs/SharedForms.md:46]: Added comment at the top referencing general documentation on the controller class

**Review by ivanELEC — APPROVED** [review]

**Pull request merged**